mungegithub/issue-labeler/simple_app.py

```diff
@@ -0,0 +1,125 @@
+import os
+import simplejson
+import logging
+from logging.handlers import RotatingFileHandler
+
+import numpy as np
+from flask import Flask, request
+from sklearn.feature_extraction import FeatureHasher
+from sklearn.externals import joblib
+from sklearn.linear_model import SGDClassifier
+from nltk.tokenize import RegexpTokenizer
+from nltk.stem.porter import PorterStemmer
+app = Flask(__name__)
+
+#Parameters
+team_fn= "./models/trained_teams_model.pkl"
+component_fn= "./models/trained_components_model.pkl"
+logFile = "/tmp/issue-labeler.log"
+logSize = 1024*1024*100
+numFeatures = 262144
+myLoss = 'hinge'
+myAlpha = .1
+myPenalty = 'l2'
+myHasher = FeatureHasher(input_type="string", n_features= numFeatures, non_negative=True)
+myStemmer = PorterStemmer()
+tokenizer = RegexpTokenizer(r'\w+')
+
+
+try:
+  stop_words = []
+  if not stopwords:
+    stop_fn = "./data/stopwords.txt"
+  with open(stop_fn, 'r') as f:
+    stopwords = [word.strip() for word in f]
+except:
+  #don't remove any stopwords
+  stopwords = []
+
+@app.errorhandler(500)
+def internal_error(exception):
+  app.logger.error(exception)
+  return "Internal Server Error-> Check the Logs"
+
+@app.route("/", methods = ["POST"])
+def get_labels():
+    title = request.form.get('title', "")
+    body = request.form.get('body', "")
+    tokens = tokenize_stem_stop(" ".join([title, body]))
+    team_mod = joblib.load(team_fn)
+    comp_mod = joblib.load(component_fn)
+    vec = myHasher.transform([tokens])
+    tlabel = team_mod.predict(vec)[0]
+    clabel = comp_mod.predict(vec)[0]
+    return ",".join([tlabel, clabel])
+
+
+def tokenize_stem_stop(inputString):
+    curTitleBody = tokenizer.tokenize(inputString.decode('utf-8').lower())
+    return map(myStemmer.stem, filter(lambda x: x not in stopwords, curTitleBody))
+
+
+@app.route("/update_models", methods = ["PUT"])
+def update_model():
+    """
+    data should contain three fields
+      titles: list of titles
+      bodies: list of bodies
+      labels: list of list of labels
+    """
+    data = request.json
+    titles = data.get('titles')
+    bodies = data.get('bodies')
+    labels = data.get('labels')
+
+    tTokens = []
+    cTokens = []
+    team_labels = []
+    component_labels = []
+    for (title, body, label_list) in zip(titles, bodies, labels):
+      tLabel = filter(lambda x: x.startswith('team'), label_list)
```

Comment:
I find it interesting that you use filter rather than list comprehension. Whatever =)